Unit12-Fruit_Search/script.js

```diff
@@ -0,0 +1,184 @@
+/*
+Title: Fruit Search project - Predictive Text
+Author: Fabricio Ribeiro
+Date: April 07, 2024
+*/
+
+//global variables
+
+const FRUIT = [
+    "Apple",
+    "Apricot",
+    "Avocado 🥑",
+    "Banana",
+    "Bilberry",
+    "Blackberry",
+    "Blackcurrant",
+    "Blueberry",
+    "Boysenberry",
+    "Currant",
+    "Cherry",
+    "Coconut",
+    "Cranberry",
+    "Cucumber",
+    "Custard apple",
+    "Damson",
+    "Date",
+    "Dragonfruit",
+    "Durian",
+    "Elderberry",
+    "Feijoa",
+    "Fig",
+    "Gooseberry",
+    "Grape",
+    "Raisin",
+    "Grapefruit",
+    "Guava",
+    "Honeyberry",
+    "Huckleberry",
+    "Jabuticaba",
+    "Jackfruit",
+    "Jambul",
+    "Juniper berry",
+    "Kiwifruit",
+    "Kumquat",
+    "Lemon",
+    "Lime",
+    "Loquat",
+    "Longan",
+    "Lychee",
+    "Mango",
+    "Mangosteen",
+    "Marionberry",
+    "Melon",
+    "Cantaloupe",
+    "Honeydew",
+    "Watermelon",
+    "Miracle fruit",
+    "Mulberry",
+    "Nectarine",
+    "Nance",
+    "Olive",
+    "Orange",
+    "Clementine",
+    "Mandarine",
+    "Tangerine",
+    "Papaya",
+    "Passionfruit",
+    "Peach",
+    "Pear",
+    "Persimmon",
+    "Plantain",
+    "Plum",
+    "Pineapple",
+    "Pomegranate",
+    "Pomelo",
+    "Quince",
+    "Raspberry",
+    "Salmonberry",
+    "Rambutan",
+    "Redcurrant",
+    "Salak",
+    "Satsuma",
+    "Soursop",
+    "Star fruit",
+    "Strawberry",
+    "Tamarillo",
+    "Tamarind",
+    "Yuzu",
+];
+
+//++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+//feature functions
+
+//This functon seaches the array of known fruits for a partial string match on any location sof the string.
+// This function returns only the 10 first match
+function search(str) {
+    const results = [];
+    const allFruits = new Set(FRUIT);
+    for (let fruit of allFruits) {
```

Comment:
```suggestion
    for (const fruit of allFruits) {
```

Use const instead of let where we don't need reassignment.

Reply:
That piece of the code is gone, but will keep this in mind.